libcamera-sys/c_api/controls.cpp

```diff
@@ -13,10 +13,87 @@ const char *libcamera_control_name(libcamera_control_id_t *control){
     return control->name().c_str();
 }
 
-enum libcamera_control_type libcamera_control_type(libcamera_control_id_t *control) {
+enum libcamera_control_type libcamera_control_id_type(libcamera_control_id_t *control) {
     return (enum libcamera_control_type) control->type();
 }
 
+const char *libcamera_control_id_vendor(libcamera_control_id_t *control) {
+    return control->vendor().c_str();
+}
+
+enum libcamera_control_direction libcamera_control_id_direction(libcamera_control_id_t *control) {
+    using Underlying = std::underlying_type_t<libcamera::ControlId::Direction>;
+    Underlying bits = static_cast<Underlying>(control->direction());
+
+    // …then cast that integer into your C enum
+    return static_cast<enum libcamera_control_direction>(bits);
+}
+
+bool libcamera_control_id_is_input(libcamera_control_id_t *control) {
+    return control->isInput();
+}
+
+bool libcamera_control_id_is_output(libcamera_control_id_t *control) {
+    return control->isOutput();
+}
+
+bool libcamera_control_id_is_array(libcamera_control_id_t *control) {
+    return control->isArray();
+}
+
+size_t libcamera_control_id_size(libcamera_control_id_t *control) {
+    return control->size();
+}
+
+size_t libcamera_control_id_enumerator_count(libcamera_control_id_t *control) {
+    return control->enumerators().size();
+}
+
+const char *libcamera_control_id_enumerator_name(libcamera_control_id_t *control, int32_t value) {
```

Comment:
`libcamera_control_id_enumerator_name` and `libcamera_control_id_enumerator_value` are never used